assets/src/scripts/monitoring-location/store/hydrograph-data.js

```diff
@@ -0,0 +1,349 @@
+                                };
+                            });
+                        });
+                    }
+                    dispatch(addMedianStatisticsData(resultStats));
+                });
+        }
+    };
+};
+
+/*
+ * Asynchronous Redux action to fetch the groundwater levels data for siteno, parameterCode, period or startTime/endTime
+ * for the dataKind
+ * @param {String} siteno
+ * @param {String} parameterCode
+ * @param {String} period - ISO 8601 duration
+ * @param {String} startTime - ISO 8601 time string
+ * @param {String} endTie - ISO 8601 time string
+ * @return {Function} that returns a Promise
+ */
+const retrieveGroundwaterLevels = function(site, {parameterCode, period, startTime, endTime}) {
+    return function(dispatch) {
+        return fetchGroundwaterLevels({site, parameterCode, period, startTime, endTime})
+            .then(data => {
+                if (data.value && data.value.timeSeries && data.value.timeSeries.length) {
+                    let values;
+                    const timeSeries = data.value.timeSeries[0];
+                    if (!timeSeries.values.length || !timeSeries.values[0].value.length) {
+                        values = [];
+                    } else {
+                        values = timeSeries.values[0].value.map((v) => {
+                            const dateTime = DateTime.fromISO(v.dateTime, {zone: 'utc'}).toMillis();
+                            return {
+                                value: parseFloat(v.value),
+                                qualifiers: v.qualifiers,
+                                dateTime: dateTime
+                            };
+
+                        });
+                    }
+                    const parameter = {
+                        parameterCode: timeSeries.variable.variableCode[0].value,
+                        name: timeSeries.variable.variableName,
+                        description: timeSeries.variable.variableDescription,
+                        unit: timeSeries.variable.unit.unitCode
+                    };
+                    dispatch(addGroundwaterLevels({
+                        parameter,
+                        values
+                    }));
+                }
+            });
+    };
+};
+
+/*
+ * Asynchronous Redux action which clears all data and then retrieves all of the data needed to display the hydrograph.
+ * The IV and groundwater levels are automatically loaded if available. Compare and median data
+ * are loaded if requested
+ * @param {String} siteno
+ * @param {String} parameterCode
+ * @param {String} period - ISO 8601 duration
+ * @param {String} startTime - ISO 8601 time string
+ * @param {String} endTime - ISO 8601 time string
+ * @param {Boolean} loadCompare
+ * @param {Boolean} loadMedian
+ * @return {Function} that returns a promise
+ */
+export const retrieveHydrographData = function(siteno, {parameterCode, period, startTime, endTime, loadCompare, loadMedian}) {
```

Comment:
This is the one that would be called whenever the time or parameter code is change.The first thing is that the data is cleared. At some point we should come up with a way to render a loading indicator. I removed the previous one which was somewhat inadequate anyways. I would like the new loading indicator to appear where the graph is. Another ticket.